src/Nest/Aggregations/Metric/WeightedAverage/WeightedAverageAggregation.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	[JsonObject(MemberSerialization = MemberSerialization.OptIn)]
+	[ContractJsonConverter(typeof(AggregationJsonConverter<WeightedAverageAggregation>))]
+	public interface IWeightedAverageAggregation : IAggregation
+	{
+		/// <summary> The configuration for the field or script that provides the values </summary>
+		[JsonProperty("value")]
+		IWeightedAverageValue Value { get; set; }
+		/// <summary> The configuration for the field or script that provides the weights /// </summary>
+		[JsonProperty("weight")]
+		IWeightedAverageValue Weight { get; set; }
+		/// <summary> The optional numeric response formatter </summary>
+		[JsonProperty("format")]
+		string Format { get; set; }
+		/// <summary> A hint about the values for pure scripts or unmapped fields </summary>
+		[JsonProperty("value_type")]
+		// TODO map as on server enum ?
+		// https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/support/ValueType.java
+		string ValueType { get; set; }
```

Comment:
Weighted Avg Aggregation is not marked as beta or experimental, so mapping as an enum might be better, as it would help guide users as to which values are supported.